client/src/library/hooks/use-delete-progress.ts

```diff
@@ -0,0 +1,47 @@
+import { useMutation } from "@tanstack/react-query";
+import { getLibraryService } from "@/domains/game/library-service";
+import { toast } from "sonner";
+import { useRouter } from "@tanstack/react-router";
+
+export const useDeleteProgress = (storyKey?: string) => {
```

Comment:
storyKey est jamais undefined en vrai, ça évitera un check inutile ligne 18

```suggestion
export const useDeleteProgress = (storyKey: string) => {
```